tests/TaskManagerTest.php

```diff
@@ -563,6 +564,21 @@ public function testTagNameNotAllowed()
         $this->taskManager->save($task1);
     }
 
+    public function testProjectHierarchyAllowed()
+    {
+        $projects = ['grandparent.parent.child', 'parent.child'];
+        foreach ($projects as $project) {
+            $task = new Task();
+            $task->setDescription('foo1');
+            $task->setProject($project);
+
+            // now exception thrown
+            $this->taskManager->save($task);
```

Comment:
Thanks, forgot about the identity map here. With or without DataProviders would probably not really save code here, but I can make the changes, sure.